### PR TITLE
Barni

### LINKED ISSUE #35: package endpoints

### PR DESCRIPTION
**Pull request by barni363hun**
fixes #35 #36 #37 #39 #40 #41

### CODE REVIEW

File: apps/api/src/app/exchange-date/exchange-date.controller.ts [hunk 1]
```diff
@@ -97,7 +94,7 @@ export class ExchangeDateController {
   @Patch()
   update(@Req() req: authRequest, @Body() body: myExchangeDateDTO) {
     return this.exchangeDateService.getById(body.id).then((a) => {
-      if (a.userId.toString() == req.user.sub) {
```
Comment (CserpakRobert) [inline apps/api/src/app/exchange-date/exchange-date.controller.ts:100]: delete ".toString()"

File: apps/api/src/app/reciever/reciever.controller.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
+import {
```
Comment (CserpakRobert) [inline apps/api/src/app/reciever/reciever.controller.ts:1]: unnecessary imports

File: apps/api/src/app/reciever/reciever.controller.ts [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  Req,
+  UseGuards,
+} from '@nestjs/common';
+import { IsEmail, IsNumber, IsPhoneNumber, IsString } from 'class-validator';
+import { AddressService } from '../address/address.service';
+import { AuthGuard, authRequest, RoleGuard } from '../auth';
+import { Roles } from '../auth/roles.decorator';
+import { RecieverService } from './reciever.service';
+
+class idDto {
```
Comment (CserpakRobert) [inline apps/api/src/app/reciever/reciever.controller.ts:19]: idDto is not used

File: apps/api/src/app/reciever/reciever.controller.ts [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  zipCode: number;
+  @IsString()
+  cityName: string;
+  @IsString()
+  street: string;
+  @IsString()
+  houseNumber: string;
+}
+
+class myRecieverDTO {
```
Comment (CserpakRobert) [inline apps/api/src/app/reciever/reciever.controller.ts:45]: myRecieverDTO is not used

File: apps/api/src/app/reciever/reciever.service.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { recieverEntity } from '../../Entities';
+import { authRequest } from '../auth';
```
Comment (CserpakRobert) [inline apps/api/src/app/reciever/reciever.service.ts:5]: imported but not used

File: apps/api/src/app/user/user.service.ts [hunk 6]
```diff
@@ -45,6 +46,32 @@ export class UserService extends GenericService<userInterface> {
...
+          data: { roles: ['rol_72IOyNuWHPWpC03o'] },
+        };
+        const data = { roles: ['rol_72IOyNuWHPWpC03o'] };
+        return this.httpService
+          .post(
+            process.env.AUTH0_DOMAIN + 'api/v2/users/' + id + '/roles',
+            data,
+            options
+          )
+          .subscribe((asd) => asd);
```
Comment (CserpakRobert) [inline apps/api/src/app/user/user.service.ts:70]: "asd" variable name

File: apps/api/src/app/user/user.service.ts [hunk 7]
```diff
@@ -84,15 +111,19 @@ export class UserService extends GenericService<userInterface> {
...
-    return this.userRepository.findOne({ id: authId }).then((res) => {
+  async getMyData(req: authRequest) {
+    return this.userRepository.findOne({ id: req.user.sub }).then((res) => {
       if (res == undefined) {
         throw new NotFoundException(
           'You can only get your data if you already have a user!'
         );
       } else {
-        return res;
+        const asd: { 'https://www.stabox.hu/roles': string[] } = {
```
Comment (CserpakRobert) [inline apps/api/src/app/user/user.service.ts:121]: "asd" variable name

File: apps/api/src/app/vehicle/vehicle.controller.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+  @Post('/all')
+  getAll() {
+    return this.vehicleService.getAll();
+  }
+
+  // gets user's all exchange date
+  @UseGuards(AuthGuard, RoleGuard)
+  @Roles('shipper')
+  @Get()
+  getMyExchangeDates(@Req() req: authRequest): Promise<vehicleEntity[]> {
```
Comment (CserpakRobert) [inline apps/api/src/app/vehicle/vehicle.controller.ts:64]: getExchangeDates returns vehicleEntity array

**Review by CserpakRobert — CHANGES_REQUESTED** [review]
"Dto" and "DTO" capitalizations are used inconsistently
".toString()" methods on user ids are unnecessary
see comments for more

**Comment by barni363hun** [comment]
@CserpakRobert please fix these problems!

**Pull request closed without merging**